File: ios/Classes/Bridges/StoreProductBridge.swift
import Flutter
import SuperwallKit

extension StoreProduct {
  func toJson() -> [String: Any] {
    var dict = [String: Any]()
    dict["productIdentifier"] = productIdentifier
    dict["localizedPrice"] = localizedPrice
    dict["localizedSubscriptionPeriod"] = localizedSubscriptionPeriod
    dict["period"] = period
    dict["periodWeeks"] = periodWeeks
    dict["periodWeeksString"] = periodWeeksString
    dict["periodMonths"] = periodMonths
    dict["periodMonthsString"] = periodMonthsString
    dict["periodYears"] = periodYears
    dict["periodYearsString"] = periodYearsString
    dict["periodDays"] = periodDays
    dict["periodDaysString"] = periodDaysString
    dict["dailyPrice"] = dailyPrice
    dict["weeklyPrice"] = weeklyPrice
    dict["monthlyPrice"] = monthlyPrice
    dict["yearlyPrice"] = yearlyPrice
    dict["hasFreeTrial"] = hasFreeTrial
    dict["trialPeriodEndDate"] = trialPeriodEndDate?.timeIntervalSince1970
    dict["trialPeriodEndDateString"] = trialPeriodEndDateString
    dict["localizedTrialPeriodPrice"] = localizedTrialPeriodPrice
    dict["trialPeriodPrice"] = trialPeriodPrice
    dict["trialPeriodDays"] = trialPeriodDays
    dict["trialPeriodDaysString"] = trialPeriodDaysString
    dict["trialPeriodWeeks"] = trialPeriodWeeks
    dict["trialPeriodWeeksString"] = trialPeriodWeeksString
    dict["trialPeriodMonths"] = trialPeriodMonths
    dict["trialPeriodMonthsString"] = trialPeriodMonthsString
    dict["trialPeriodYears"] = trialPeriodYears
    dict["trialPeriodYearsString"] = trialPeriodYearsString
    dict["trialPeriodText"] = trialPeriodText
    dict["locale"] = locale
    dict["languageCode"] = languageCode
    dict["currencyCode"] = currencyCode
    dict["currencySymbol"] = currencySymbol
    dict["isFamilyShareable"] = isFamilyShareable
    dict["regionCode"] = regionCode
    dict["price"] = price
    return dict
  }
}

File: android/src/main/kotlin/com/superwall/superwallkit_flutter/bridges/StoreProductBridge.kt
package com.superwall.superwallkit_flutter.bridges

import com.superwall.sdk.store.abstractions.product.StoreProduct

fun StoreProduct.toJson(): Map<String, Any?> {
    return mapOf(
        "productIdentifier" to productIdentifier,
        "localizedPrice" to localizedPrice,
        "localizedSubscriptionPeriod" to localizedSubscriptionPeriod,
        "period" to period,
        "periodWeeks" to periodWeeks,
        "periodWeeksString" to periodWeeksString,
        "periodMonths" to periodMonths,
        "periodMonthsString" to periodMonthsString,
        "periodYears" to periodYears,
        "periodYearsString" to periodYearsString,
        "periodDays" to periodDays,
        "periodDaysString" to periodDaysString,
        "dailyPrice" to dailyPrice,
        "weeklyPrice" to weeklyPrice,
        "monthlyPrice" to monthlyPrice,
        "yearlyPrice" to yearlyPrice,
        "hasFreeTrial" to hasFreeTrial,
        "trialPeriodEndDate" to trialPeriodEndDate?.time,
        "trialPeriodEndDateString" to trialPeriodEndDateString,
        "localizedTrialPeriodPrice" to localizedTrialPeriodPrice,
        "trialPeriodDays" to trialPeriodDays,
        "trialPeriodDaysString" to trialPeriodDaysString,
        "trialPeriodWeeks" to trialPeriodWeeks,
        "trialPeriodWeeksString" to trialPeriodWeeksString,
        "trialPeriodMonths" to trialPeriodMonths,
        "trialPeriodMonthsString" to trialPeriodMonthsString,
        "trialPeriodYears" to trialPeriodYears,
        "trialPeriodYearsString" to trialPeriodYearsString,
        "trialPeriodText" to trialPeriodText,
        "locale" to locale,
        "languageCode" to languageCode,
        "currencyCode" to currencyCode,
        "currencySymbol" to currencySymbol,
        "regionCode" to regionCode,
        "price" to price
    )
}
